src/gui/DatabaseWidget.cpp

```diff
@@ -1520,13 +1524,34 @@ void DatabaseWidget::onGroupChanged()
 
 void DatabaseWidget::onDatabaseModified()
 {
-    if (!m_blockAutoSave && config()->get(Config::AutoSaveAfterEveryChange).toBool()) {
+    refreshSearch();
+    int autosaveDelay = m_db->metadata()->autosaveDelay() * 60 * 1000; // min to msec for QTimer
+    bool autosaveAfterEveryChangeConfig = config()->get(Config::AutoSaveAfterEveryChange).toBool();
+    if (m_autosaveDelayTimer->isActive()) {
+        // reset delay when modified
+        m_autosaveDelayTimer->start(autosaveDelay);
+        return;
+    }
+    if (autosaveDelay > 0 && autosaveAfterEveryChangeConfig) {
```

Comment:
Is using the `DefaultAutosaveDelay` constant better that hard-coded 0?

Reply:
> Is using the `DefaultAutosaveDelay` constant better that hard-coded 0?

The default is 0 at the moment but might be changed in the future. If that's the case this function would break, the function checks if the delay exists or not therefor its better to check for hard-coded 0 value.

Reply:
Right, it makes sense.